loudcloud-dev/ee_calculator
# feat: leaves edit/cancel and index page (part 1)

## Changelog

### Features

#### Controller
- added standard `edit` and `update` methods
- added `cancel` method
    - patches leave `status` to "cancelled" 

#### View
- added `edit` view (reused create form)
- improved `index` view (still incomplete)
    - added bootstrap classes
    - edit button only appears on pending leaves
    - cancel button only appears on pending and approved leaves
    - missing features (for next PR)
        - display remaining leaves
        - filters (`leave_type`, `status`, date range, etc)
        - colors for leave type
        - cancel confirmation modal

### Fixes

#### Model
- added `optional: true` to `processed_by`
 
## How to check
- log in to an employee account and file leaves
- verify new "My Leaves" UI
- verify that pending leaves can be edited successfully
- verify that pending and approved leaves can be cancelled successfully
